src/views/index.pug

```diff
@@ -22,4 +22,46 @@ preview
         label(for='comments') Comments?
         textarea(id='comments' name='comments').form-control
       button(type='submit').btn.btn-primary Submit
+script(type='text/javascript')
+  | var tickingElements = [];
+  | function hookElement(id) {
+  |   var element = document.getElementById(id);
+  |   var elementObject = {
+  |     hasValue: false,
+  |   };
+  |   tickingElements.push(elementObject);
```

Comment:
Why do you need `elementObject`? Can't you declare `var hasValue`, and push that to ticking elements?

Reply:
@mxmeinhold if I do that, I can't later mutate it unless I assume the index will stay consistent (not really a safe assumption when using an array, I'd rather not do that)

Reply:
Isn't `hasValue` in the scope of `hookElement`? Can't you use this as a closure? Or, since you're pushing it to tickingElements, do only objects get passed by reference?

Reply:
Right, `tickingElements` is the issue here. Only objects get passed by reference.